homeassistant/bootstrap.py

```diff
@@ -325,15 +325,38 @@ def _get_domains(hass: core.HomeAssistant, config: Dict[str, Any]) -> Set[str]:
     hass: core.HomeAssistant, config: Dict[str, Any]
 ) -> None:
     """Set up all the integrations."""
+
+    async def async_setup_multi_components(domains: Set[str]) -> None:
+        """Set up multiple domains. Log on failure."""
+        futures = {
+            domain: hass.async_create_task(async_setup_component(hass, domain, config))
+            for domain in domains
+        }
+        await asyncio.wait(futures.values())
+        errors = [
+            domain
+            for domain in domains
+            if futures[domain].exception() or not futures[domain].result()
+        ]
+        for domain in errors:
+            if futures[domain].exception():
+                _LOGGER.error(
+                    "Error setting up integration %s - received exception %s",
+                    domain,
+                    str(futures[domain].exception()),
```

Comment:
If this is an exception instance, it has a magic method `__str__` and we don't need to copy to string.

Reply:
fixed